src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -206,6 +206,31 @@ public SourceColumnMetadata(ValueMethod method, bool isSqlType, bool isDataFeed)
         private DataRowState _rowStateToSkip;
         private IEnumerator _rowEnumerator;
 
+        private int RowNumber
+        {
+            get
+            {
+                int rowNo;
+
+                switch (_rowSourceType)
+                {
+                    case ValueSourceType.RowArray:
+                        rowNo = ((DataTable)_dataTableSource).Rows.IndexOf(_rowEnumerator.Current as DataRow);
+                        break;
+                    case ValueSourceType.DataTable:
+                        rowNo = ((DataTable)_rowSource).Rows.IndexOf(_rowEnumerator.Current as DataRow);                        
+                        break;
+
+                    case ValueSourceType.DbDataReader:
+                    case ValueSourceType.IDataReader:
+                    case ValueSourceType.Unspecified:
+                    default:
+                        return -1;                        
+                }
+                return ++rowNo;
```

Comment:
Why do we increment the number?

Reply:
The **row number** is new information that is added to the error message. We have just decided to start it by **1** instead of **0**. It has some pros and cons. Please leave your idea to make the best decision.